include/fmt/base.h

```diff
@@ -270,6 +270,16 @@ FMT_PRAGMA_CLANG(diagnostic push)
 #  define FMT_WIN32 0
 #endif
 
+#if (FMT_GCC_VERSION || FMT_CLANG_VERSION) && !defined(FMT_HEADER_ONLY) && \
+    !FMT_WIN32
+// Header only (which implies everything is inline) and weak symbols do not work
+// together. And on Windows weak symbols seem not to work properly.
+#  define FMT_WEAK __attribute__((weak))
+#endif
+#ifndef FMT_WEAK
+#  define FMT_WEAK
+#endif
```

Comment:
Looking at all the caveats, I am now leaning towards your previous version with a configuration macro.

Also what's up with the test failures?